akka-docs/src/main/paradox/stream/stream-refs.md

```diff
@@ -0,0 +1,179 @@
+send to the *origin* side data in a streaming, flow-controlled fashion. The origin here allocates a Sink,
+which could be as simple as a `Sink.foreach` or as advanced as a complex sink which streams the incoming data
+into various other systems (e.g. any of the Alpakka provided Sinks).
+
+@@@ note
+  To form a good mental model of `SinkRef`s, you can think of them as being similar to "passive mode" in FTP.
+@@@
+
+Scala
+:   @@snip [FlowStreamRefsDocSpec.scala]($code$/scala/docs/stream/FlowStreamRefsDocSpec.scala) { #offer-sink }
+
+Java
+:   @@snip [FlowStreamRefsDocTest.java]($code$/java/jdocs/stream/FlowStreamRefsDocTest.java) { #offer-sink }
+
+Using the offered `SinkRef` to send data to the origin of the Sink is also simple, as we can treat the 
+SinkRef just as any other Sink and directly `runWith` or `run` with it.
+
+Scala
+:   @@snip [FlowStreamRefsDocSpec.scala]($code$/scala/docs/stream/FlowStreamRefsDocSpec.scala) { #offer-sink-use }
+
+Java
+:   @@snip [FlowStreamRefsDocTest.java]($code$/java/jdocs/stream/FlowStreamRefsDocTest.java) { #offer-sink-use }
+
+The process of preparing and running a `SinkRef` powered distributed stream is shown by the animation below: 
+
+![SourceRef usage animation](../images/sink-ref-animation.gif)
+
+
+@@@ warning
+  A `SinkeRef` is *by design* "single-shot". i.e. it may only be materialized once.
+  This is in order to not complicate the mental model what materializing such value would mean.
+  
+  If you have an use case for building a fan-in operation accepting writes from multiple remote nodes,
+  you can build your Sink and prepend it with a `Merge` stage, each time materializing a new `SinkRef`
+  targeting that Merge. This has the added benefit of giving you full control how to merge these streams 
+  (i.e. by using "merge preferred" or any other variation of the fan-in stages).
+@@@
+
+## Bulk Stream References
+
+@@@ warning
+  Bulk stream references are not implemented yet.
+  See ticket [Bulk Transfer Stream Refs #24276](https://github.com/akka/akka/issues/24276) to track progress or signal demand for this feature.  
+@@@
+
+Bulk stream refs can be used to create simple to use side-channels to transfer humongous amounts 
+of data such as huge log files, messages or even media, with as much ease as if it was a trivial local stream.
+
+## Configuration
+
+### Stream reference subscription timeouts
+
+All stream references have a subscription timeout, which is intended to prevent resource leaks
+in situations in which a remote node would requests the allocation of many streams yet never actually run
+them. In order to prevent this, each stream reference has a default timeout (of 30 seconds), after which
+if it's "handed out" side has not been materialized, the origin will terminate with a timeout exception,
```

Comment:
=> "after which the origin will abort the stream offer if the target has not materialized the stream ref in time. After the timeout has triggered, materialization of the target side will fail pointing out that the origin is missing."